RdlMigration/ConvertRDL.cs

```diff
@@ -560,6 +577,50 @@ private XElement CreateDataSourceNode(XDocument doc, DataSourceDefinition curren
             return currNode;
         }
 
+        private void DiscoverSubreports(XDocument doc)
+        {
+            var subreports = doc.Descendants(doc.Root.Name.Namespace + "Subreport");
+            var subreportPaths = new List<string>();
+            foreach (XElement subreport in subreports)
+            {
+                var subreportName = subreport.Elements().First().Value;
+                var subreportPath = Path.Combine(rootFolder, subreportName);
+                subreportPath = Path.GetFullPath(subreportPath).Replace("\\", "/");
+                subreportPath = subreportPath.Substring(subreportPath.IndexOf('/'));
+                subreportName = Path.GetFileName(subreportPath);
+                
+                try
+                {
+                    if (rdlFileIO.IsFolder(subreportPath))
```

Comment:
This is a weird check, you should be more explicit here and add a method that checks this is a report. Also, would a subreport be able to point to a linked report? if so, what do we expect the behavior to be for that case.

Reply:
agreed that this is a weird check, will add new function. what is a linked report?

Reply:
https://docs.microsoft.com/en-us/sql/reporting-services/reports/create-a-linked-report?view=sql-server-ver15

Basically it's a shadow copy of an existing rdl report in server.

Reply:
i tested with rose reports, seems like we're able to point to linked reports as subreports. however, the rdlmigration tool does not recognize linked reports as actual files and will not migrate their actual file. do we want to add support for this? not just for subreports. @gboreki

Reply:
not adding support for now. if customers complain about this we can add it in the future.